lib/opts.nit

```diff
@@ -147,12 +147,25 @@ abstract class OptionParameter
 	protected fun convert(str: String): VALUE is abstract
 
 	# Is the parameter mandatory?
-	var parameter_mandatory: Bool = true is writable
+	var parameter_mandatory = true is writable
 
-	redef fun read_param(it)
+	redef fun read_param(opts, it)
 	do
 		super
-		if it.is_ok and (it.item.is_empty or it.item.chars.first != '-') then
+
+		var ok = it.is_ok
+		if ok and not it.item.is_empty and it.item.chars.first == '-' then
+			# The next item may look like a known command
+			for opt in opts.options do
+				if opt.names.has(it.item) then
+					# The next item is a known command
+					ok = false
+					break
+				end
+			end
+		end
```

Comment:
I can remove that feature, it bothers me too.

Do I remove the mandatory_parameter too? It allowed for an OptionString to have no parameter if it was followed by another parameter: `nitc --compile-dir -o a.out` where compile-dir would have a default value. I will be pretty much useless if it can only apply at the end of the command. There is no user of this feature in the repo. I could also keep the current behavior only when `mandatory_parameter == false`.